### DISCUSSION TITLE
Download entire contents of the local Wiki.js installation, e.g. in markdown

### DISCUSSION THREAD

**Discussion by tshtatland** [requarks/wiki#5543] [Help / Questions] [answered]
I am considering porting all pages of a local Wiki.js installation to another system. An easy option could be downloading all the contents in markdown. How can I do this?

Note: I found that in the postgreSQL database, the table `pages` has the text of the wiki pages. The table `assets` has the names of the image files. But I could not quickly find where the image files themselves are stored, e.g., the file paths.

**Comment by NGPixel** [comment, marked as answer]
You need to enable the "Local File System" storage module and then do a "Dump all content to disk".

**Reply by tshtatland** [reply]
Thank you for the help! Sorry for the duplicate postings here and on Slack. Moving the discussion from Slack back here, where it is easier to post and search.

I enabled permissions like so on the server where I installed Wiki.js:
```
mkdir -p /tmp/wikijs
chmod a+rwx /tmp/wikijs
```
I set:
`local file system: active`, `storage path: /tmp/wikijs`.  
Then `Apply`.

But when I do any of these:
`Dump all content to disk [run]` or `Create Backup [run]`, I find no contents in the local storage directory. I expected a backup file. What am I doing wrong?

I also followed another suggestion (from Slack): use Export utility (under Administration Area > Utilities). I used as target folder path a directory that does not exist yet: `/tmp/wikijs/test` under an existing directory `/tmp/wikijs` with permissions for all to read/write (see above). I selected the items to export (everything), then `Start export`. The UI task completed without warnings/errors, but there was no result on the disk. Directory `/tmp/wikijs` was empty.

**Reply by NGPixel** [reply]
Are you sure you're looking in the right location? Are you using a docker container?

**Reply by tshtatland** [reply]
Found it in the docker! Thank you.
```
$ docker exec -it wiki bash
ls -lrt /tmp/wikijs/
total 72
drwxr-xr-x    2 node     node          4096 Aug 17 17:38 galaxy
-rw-r--r--    1 node     node           423 Aug 17 17:40 home.md
-rw-r--r--    1 node     node         10272 Aug 17 17:40 git.md
-rw-r--r--    1 node     node         31871 Aug 17 17:40 delete_multiple_histories.png
-rw-r--r--    1 node     node          3599 Aug 17 17:40 best-practices.md
drwxr-xr-x    2 node     node          4096 Aug 17 17:40 _manual
drwxr-xr-x    3 node     node          4096 Aug 17 19:31 test
drwxr-xr-x    2 node     node          4096 Aug 18 17:39 _daily
drwxr-xr-x    3 node     node          4096 Aug 18 19:32 test_folder1
bash-5.1$ ls -lrt /tmp/wikijs/test
total 64
-rw-r--r--    1 node     node           641 Aug 17 19:31 users.json.gz
-rw-r--r--    1 node     node         18558 Aug 17 19:31 settings.json
-rw-r--r--    1 node     node          9378 Aug 17 19:31 pages.json.gz
-rw-r--r--    1 node     node         14745 Aug 17 19:31 pages-history.json.gz
-rw-r--r--    1 node     node           365 Aug 17 19:31 navigation.json
-rw-r--r--    1 node     node          1984 Aug 17 19:31 groups.json
drwxr-xr-x    2 node     node          4096 Aug 17 19:31 assets
```
